homeassistant/components/livisi/climate.py

```diff
@@ -0,0 +1,211 @@
+    config_entry: ConfigEntry,
+    device: dict[str, Any],
+    coordinator: LivisiDataUpdateCoordinator,
+) -> ClimateEntity:
+    """Create Climate Entity."""
+    capabilities: Mapping[str, Any] = device[CAPABILITY_MAP]
+    room_id: str = device["location"]
+    room_name: str = coordinator.rooms[room_id]
+    livisi_climate = LivisiClimate(
+        config_entry,
+        coordinator,
+        unique_id=device["id"],
+        manufacturer=device["manufacturer"],
+        device_type=device["type"],
+        target_temperature_capability=capabilities["RoomSetpoint"],
+        temperature_capability=capabilities["RoomTemperature"],
+        humidity_capability=capabilities["RoomHumidity"],
+        room=room_name,
+    )
+    return livisi_climate
+
+
+class LivisiClimate(CoordinatorEntity[LivisiDataUpdateCoordinator], ClimateEntity):
+    """Represents the Livisi Climate."""
+
+    def __init__(
+        self,
+        config_entry: ConfigEntry,
+        coordinator: LivisiDataUpdateCoordinator,
+        unique_id: str,
+        manufacturer: str,
+        device_type: str,
+        target_temperature_capability: str,
+        temperature_capability: str,
+        humidity_capability: str,
+        room: str,
+    ) -> None:
+        """Initialize the Livisi Climate."""
+        self.config_entry = config_entry
+        self._attr_unique_id = unique_id
+        self._target_temperature_capability = target_temperature_capability
+        self._temperature_capability = temperature_capability
+        self._humidity_capability = humidity_capability
+        self.aio_livisi = coordinator.aiolivisi
+        self._attr_available = False
+        self._attr_hvac_modes = [HVACMode.HEAT]
+        self._attr_hvac_mode = HVACMode.HEAT
+        self._attr_temperature_unit = UnitOfTemperature.CELSIUS
+        self._attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
+        self._attr_target_temperature_high = MAX_TEMPERATURE
+        self._attr_target_temperature_low = MIN_TEMPERATURE
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, unique_id)},
+            manufacturer=manufacturer,
+            model=device_type,
+            name=room,
+            suggested_area=room,
+            via_device=(DOMAIN, config_entry.entry_id),
+        )
+        super().__init__(coordinator)
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        response = await self.aio_livisi.async_vrcc_set_temperature(
+            self._target_temperature_capability,
+            kwargs.get(ATTR_TEMPERATURE),
+            self.coordinator.is_avatar,
+        )
+        if response is None:
+            self._attr_available = False
+            raise HomeAssistantError(f"Failed to turn off {self._attr_name}")
+
+    def set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Do nothing as LIVISI devices do not support changing the hvac mode."""
```

Comment:
I don't think it should do nothing as that will not tell the users why nothing is happening.

I think you should instead raise a `HomeAssistantError` to indicate that it is not supported.

Reply:
I raised this error, thank you.

Reply:
Should I open a new pull request to change this?

Reply:
Yea please

Reply:
The new pull request is here: https://github.com/home-assistant/core/pull/88844